java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java

```diff
@@ -31,18 +41,61 @@ public MenuControllerImpl(MenuService menuService) {
                                                        final Integer pageSize,
                                                        final Integer pageNumber) {
 
+        List<SearchMenuResultItem> menuResultItems;
+        if (StringUtils.isNotEmpty(searchTerm) && Objects.nonNull(restaurantId) ) {
+            return ResponseEntity.ok(
+                    new SearchMenuResult(pageSize, pageNumber, getResultsByRestaurantIdAndNameContaining(searchTerm,restaurantId, pageSize, pageNumber)));
+        }
+
+        if (StringUtils.isNotEmpty(searchTerm)) {
```

Comment:
Only one of these if's would be true so use if/else
Also, having so many return statements in a method is confusing and bad practice.